perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java

```diff
@@ -106,12 +110,14 @@ private GenResourceDataNode getDataForResource(Resource resource) {
 		} else {
 			members = sess.getPerunBl().getResourcesManagerBl().getAllowedMembers(sess, resource);
 		}
-		if (BeansUtils.getCoreConfig().getForceConsents()) {
-			// remove the members without granted consents on required attributes
-			members.removeIf(member -> !membersWithConsent.contains(member));
-		} else {
-			// we skipped this part if consents were required, so add them now
-			membersWithConsent.addAll(members);
+		if (consentEval) {
```

Comment:
Join into one IF statement. We still want to run the code in this else block if consentEval is false.